packages/upload-api/src/types/usage.ts

```diff
@@ -13,4 +15,10 @@ export interface UsageStorage {
     space: SpaceDID,
     period: { from: Date; to: Date }
   ) => Promise<Result<UsageData, Failure>>
+  record: (
+    customer: AccountDID,
+    resource: UnknownLink,
```

Comment:
Would be good to get a comment here - what is the resource? It's the root CID of a DAG (not the CID of a CAR) or I imagine it could also be literally a CID of any sub-DAG within a DAG.

Reply:
It is the same data CID we use in the gateway to fetch the content. It is pulled from the URL via `withParsedIpfsUrl` handler from the `gateway-lib`: https://github.com/storacha/gateway-lib/blob/main/src/middleware.js#L118
I think this is the Root CID of a DAG, but I am not 100% sure. Do you know?

Reply:
https://w3s.link/ipfs/bafybeibrqc2se2p3k4kfdwg7deigdggamlumemkiggrnqw3edrjosqhvnm/yesthisisdog.jpg

In the above URL, the root CID is `bafybeibrqc2se2p3k4kfdwg7deigdggamlumemkiggrnqw3edrjosqhvnm`.

If you look at the HTTP headers, the etag will tell you the CID of the resource at the end of the path i.e. the CID of `yesthisisdog.jpg`, which is `bafkreiem4twkqzsq2aj4shbycd4yvoj2cx72vezicletlhi7dijjciqpui`.

You can use this as a root CID too!
https://w3s.link/ipfs/bafkreiem4twkqzsq2aj4shbycd4yvoj2cx72vezicletlhi7dijjciqpui

If you look at the HTTP headers, the etag is `bafkreiem4twkqzsq2aj4shbycd4yvoj2cx72vezicletlhi7dijjciqpui`, since there is no path.

So when I say "root" CID I mean the CID that might be considered the root CID of a given upload. i.e. a CID that would be set as `root` in an `upload/add` invocation.

All I think is needed is a clarification that the `resource` as you have it there is _not_ the CID of a blob/shard, it's the CID of some gateway accessible content.